needle.pm

```diff
@@ -156,10 +138,7 @@ sub unregister ($self, $reason = undef) {
 sub register ($self) {
     my %check_dups;
     for my $g (@{$self->{tags}}) {
-        if ($check_dups{$g}) {
-            bmwqemu::diag("$self->{name} contains $g twice");
-            next;
-        }
+        bmwqemu::diag("$self->{name} contains $g twice"), next if ($check_dups{$g});
```

Comment:
Have you tried to cover this?
If something is not too difficult to cover, we should do it. The ideal goal is to have 100% coverage, not to appear every line as covered although it really isn't.
I know I disagree with @okurz here but I think we shouln't overuse this one-line trick.

Reply:
And I must also say that using the comma operator like this is probably not a good idea as it makes the code less readable (and is it even correct?).